pkg/cloud/azure/services/network/publicipaddress.go

```diff
@@ -25,7 +25,23 @@ import (
 	clusterv1 "sigs.k8s.io/cluster-api/pkg/apis/cluster/v1alpha1"
 )
 
-// CreateOrUpdatePublicIPAddress retrieves the Public IP address resource.
+// GetPublicIPAddress retrieves the Public IP address resource.
+func (s *Service) GetPublicIPAddress(resourceGroup, IPName string) (network.PublicIPAddress, error) {
+	pip, err := s.scope.PublicIPAddresses.Get(
+		s.scope.Context,
+		resourceGroup,
+		IPName,
+		"",
+	)
+
+	if err != nil {
```

Comment:
lines 37-42 are redundant. We can just do `return pip, nil` instead.

Reply:
or even return `scope.PublicIPAddresses.Get(s.scope.Context, resourceGroup, IPName, "")` .

Reply:
Thanks for catching. I've wrapped the error (which is what I meant to do the first time around. :sweat_smile:)